File: src/main/java/gargoyle/sexbomb/util/applet/JApplet.java
package gargoyle.sexbomb.util.applet;

import javax.swing.JMenuBar;
import javax.swing.JPanel;
import java.awt.Image;
import java.io.InputStream;
import java.net.URL;
import java.util.Enumeration;
import java.util.Iterator;

public class JApplet extends JPanel implements Applet {
    private final AppletContext context = new AppletContextImpl();
    private AppletStub stub;

    public void appletResize(int width, int height) {
        stub.appletResize(width, height);
    }

    public Applet getApplet(String name) {
        return stub.getApplet(name);
    }

    public AppletContext getAppletContext() {
        return context;
    }

    public Enumeration<Applet> getApplets() {
        return stub.getApplets();
    }

    public AudioClip getAudioClip(URL url) {
        return stub.getAudioClip(url);
    }

    public URL getCodeBase() {
        return stub.getCodeBase();
    }

    public URL getDocumentBase() {
        return stub.getDocumentBase();
    }

    public Image getImage(URL url) {
        return stub.getImage(url);
    }

    public String getParameter(String name) {
        return stub.getParameter(name);
    }

    public InputStream getStream(String key) {
        return stub.getStream(key);
    }

    public Iterator<String> getStreamKeys() {
        return stub.getStreamKeys();
    }

    public void setStream(String key, InputStream stream) {
        stub.setStream(key, stream);
    }

    @Override
    public void setStub(AppletStub stub) {
        this.stub = stub;
    }

    @Override
    public void init() {
    }

    @Override
    public void start() {
    }

    @Override
    public void stop() {
    }

    @Override
    public void destroy() {
    }

    protected void showDocument(URL url) {
        stub.showDocument(url);
    }

    protected void showDocument(URL url, String target) {
        stub.showDocument(url, target);
    }

    protected void showStatus(String status) {
        stub.showStatus(status);
    }

    protected void setJMenuBar(JMenuBar bar) {
        super.getRootPane().setJMenuBar(bar);
    }
}

File: src/main/kotlin/gargoyle/sexbomb/util/applet/JApplet.kt
package gargoyle.sexbomb.util.applet

import java.awt.Image
import java.io.InputStream
import java.net.URL
import java.util.Enumeration
import javax.swing.JMenuBar
import javax.swing.JPanel

open class JApplet : JPanel(), Applet {
    val appletContext: AppletContext = AppletContextImpl()
    private lateinit var stub: AppletStub
    fun appletResize(width: Int, height: Int) {
        stub.appletResize(width, height)
    }

    fun getApplet(name: String): Applet {
        return stub.getApplet(name)
    }

    val applets: Enumeration<Applet>
        get() = stub.applets

    fun getAudioClip(url: URL): AudioClip {
        return stub.getAudioClip(url)
    }

    val codeBase: URL?
        get() = stub.codeBase
    val documentBase: URL?
        get() = stub.documentBase

    fun getImage(url: URL): Image {
        return stub.getImage(url)
    }

    fun getParameter(name: String): String? {
        return stub.getParameter(name)
    }

    fun getStream(key: String): InputStream? {
        return stub.getStream(key)
    }

    val streamKeys: Iterator<String>
        get() = stub.streamKeys

    fun setStream(key: String, stream: InputStream) {
        stub.setStream(key, stream)
    }

    override fun setStub(stub: AppletStub) {
        this.stub = stub
    }

    override fun init() {}
    override fun start() {}
    override fun stop() {}
    override fun destroy() {}
    protected open fun showDocument(url: URL) {
        stub.showDocument(url)
    }

    protected open fun showDocument(url: URL, target: String) {
        stub.showDocument(url, target)
    }

    protected fun showStatus(status: String) {
        stub.showStatus(status)
    }

    protected fun setJMenuBar(bar: JMenuBar) {
        super.getRootPane().jMenuBar = bar
    }
}
